### ISSUE TITLE
plugin shows as broken when trying to install

### ISSUE THREAD

**Issue by rful011** [enricofer/changeDataSource#7]
HI

I am on a Mac running 10.12.04 and QGIS 2.14.3-Essen.  When I try and install the changedatasource plugin I get an error.  Here is the details:

Couldn't load plugin changeDataSource due to an error when calling its classFactory() method 

TypeError: 'QPyNullVariant' object has no attribute '__getitem__' 
Traceback (most recent call last):
  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 306, in startPlugin
    plugins[packageName] = package.classFactory(iface)
  File "/Users/rful011/.qgis2/python/plugins/changeDataSource/__init__.py", line 35, in classFactory
    return changeDataSource(iface)
  File "/Users/rful011/.qgis2/python/plugins/changeDataSource/changeDataSource.py", line 52, in __init__
    locale = QSettings().value('locale/userLocale')[0:2]
TypeError: 'QPyNullVariant' object has no attribute '__getitem__'

Python version: 2.7.10 (default, Feb  6 2017, 23:53:20) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] 
QGIS version: 2.14.3-Essen Essen, exported 

Python Path:
/Users/rful011/.qgis2/python/plugins/processing
/Applications/QGIS.app/Contents/MacOS/../Resources/python
/Users/rful011/.qgis2/python
/Users/rful011/.qgis2/python/plugins
/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins
/Library/Frameworks/SQLite3.framework/Versions/C/Python/2.7
/Library/Frameworks/GEOS.framework/Versions/3/Python/2.7/site-packages
/Library/Python/2.7/site-packages/numpy-override
/Library/Python/2.7/site-packages/matplotlib-override
/Library/Frameworks/MapServer.framework/Versions/6.0/Python/2.7
/Users/rful011/.qgis2/python/plugins/opengeo/ext-libs/httplib2-0.8-py2.7.egg
/Users/rful011/.qgis2/python/plugins/opengeo/ext-libs/requests-2.0.1-py2.7.egg
/Users/rful011/.qgis2/python/plugins/opengeo/ext-libs/raven-3.5.1-py2.7.egg
/Users/rful011/.qgis2/python/plugins/opengeo/ext-libs/gsconfig-0.6.6-py2.7.egg
/Users/rful011/.qgis2/python/plugins/opengeo/ext-libs/gsimporter-0.1-py2.7.egg
/Users/rful011/.qgis2/python/plugins/opengeo/ext-libs/nose-1.3.0-py2.7.egg
/Users/rful011/.qgis2/python/plugins/opengeo/ext-libs/nose_html-1.1-py2.7.egg
/Users/rful011/.qgis2/python/plugins/opengeo/ext-libs/coverage-3.7-py2.7-win32.egg
/Library/Frameworks/GDAL.framework/Versions/1.11/Python/2.7/site-packages
/Library/Python/2.7/site-packages/distribute-0.6.28-py2.7.egg
/Library/Python/2.7/site-packages/matplotlib-1.4.x-py2.7-macosx-10.9-intel.egg
/Library/Python/2.7/site-packages/mock-1.0.1-py2.7.egg
/Library/Python/2.7/site-packages/nose-1.3.0-py2.7.egg
/Library/Python/2.7/site-packages/pyparsing-2.0.1-py2.7.egg
/Library/Python/2.7/site-packages/tornado-3.2-py2.7-macosx-10.9-intel.egg
/Library/Python/2.7/site-packages/six-1.5.2-py2.7.egg
/Library/Python/2.7/site-packages/backports.ssl_match_hostname-3.4.0.2-py2.7.egg
/Library/Python/2.7/site-packages/pip-1.5.4-py2.7.egg
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python27.zip
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-darwin
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac/lib-scriptpackages
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-tk
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-old
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload
/Library/Python/2.7/site-packages
/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python
/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/PyObjC
/Users/rful011/.qgis2//python
/Users/rful011/.qgis2/python/plugins/opengeo/ext-libs
/Users/rful011/src/qgis/scripts/test
/Applications/QGIS.app/Contents/Resources/python/plugins/fTools/tools
/Users/rful011/Tiri maps/qgis-projects

**Comment by enricofer** [comment]
Hi Russel,
Looking to the trace appears that the plugin is getting error reading a locale setting. This is very strange and makes me think to a bad installation because this row is common to many plugins expecially those made with a standard template. Did you try to install another plugin? Do you get the same?

**Comment by rful011** [comment]
> On 4/05/2017, at 6:26 PM, enrico ferreguti <notifications@github.com> wrote:
> 
> Hi Russel,
> Looking to the trace appears that the plugin is getting error reading a locale setting. This is very strange and makes me think to a bad installation because this row is common to many plugins expecially those made with a standard template. Did you try to install another plugin? Do you get the same?
> 
> 

Hi Enrico

I have just successfully installed two other plugins but that does not mean the the issues are not with my python env.   I recently tried to set up a plugin development environment on my Mac and ran into all sorts of issues related python packages — some of which would only install versions compatible with python 3.  I also tripped over Apple’s new SIP scheme that prevents files in certain directories being modified even if you are root.  At which point I gave up and decided to wait until QGIS 3 where I can do everything in python 3.

At some point I am going to need to rebuild my Mac from scratch and that is going to be painful.  There is five years worth of development tools and other stuff.  

Feel free to close the issue as being “user related” ; )
